client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -41,51 +41,33 @@ const siteMigration: Flow = {
 			? [ STEPS.PICK_SITE, STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ]
 			: [];
 
-		return [ ...baseSteps, ...hostedVariantSteps ];
+		return stepsWithRequiredLogin( [ ...baseSteps, ...hostedVariantSteps ] );
 	},
 
 	useAssertConditions(): AssertConditionResult {
 		const { siteSlug, siteId } = useSiteData();
+		const { isOwner } = useIsSiteOwner();
 		const userIsLoggedIn = useSelect(
 			( select ) => ( select( USER_STORE ) as UserSelect ).isCurrentUserLoggedIn(),
 			[]
 		);
-		const startUrl = useStartUrl( this.variantSlug ?? FLOW_NAME );
-
-		let result: AssertConditionResult = { state: AssertConditionState.SUCCESS };
-		const { isOwner } = useIsSiteOwner();
-
-		useEffect( () => {
-			if ( ! userIsLoggedIn ) {
-				const logInUrl = startUrl;
-				window.location.assign( logInUrl );
-			}
-		}, [ startUrl, userIsLoggedIn ] );
 
 		useEffect( () => {
-			if ( isOwner === false ) {
+			if ( isOwner === false && userIsLoggedIn ) {
 				window.location.assign( '/start' );
 			}
-		}, [ isOwner ] );
-
-		if ( ! userIsLoggedIn ) {
-			result = {
-				state: AssertConditionState.FAILURE,
-				message: 'site-migration requires a logged in user',
-			};
-
-			return result;
-		}
+		}, [ isOwner, userIsLoggedIn ] );
 
 		if ( ! siteSlug && ! siteId && ! isHostedSiteMigrationFlow( this.variantSlug ?? FLOW_NAME ) ) {
 			window.location.assign( '/start' );
-			result = {
+
+			return {
 				state: AssertConditionState.FAILURE,
 				message: 'site-setup did not provide the site slug or site id it is configured to.',
 			};
 		}
```

Comment:
I am wondering if we have two issues in this code block:
1. This should be inside `useEffect()`
2. Should this code also be checking for whether we have a logged-in user?

Reply:
As a follow-up, I think this is another situation where we had an implicit dependency on the login redirect logic, which we've now removed.

Reply:
Fix pushed in 4614bf4de111ec61e5717f2782630594c2b32cfb.